Update Fleet settings for 7.14 (#103380)
elastic/kibana: docs/settings/fleet-settings.asciidoc

  | Hostnames used by {agent} for accessing {fleet-server}.
| `xpack.fleet.agents.elasticsearch.hosts`
  | Hostnames used by {agent} for accessing {es}.
| `xpack.fleet.agents.elasticsearch.ca_sha256`
  | Hash pin used for certificate verification. The pin is a base64-encoded
    string of the SHA-256 fingerprint.
|===

//QUESTION: Is the xpack.fleet.agents.fleet_server.hosts setting supported? It

> //wasn't listed in https://github.com/elastic/observability-docs/issues/758.

Reviewer: yes it is, I missed that in the original issue

> //QUESTION: Is the xpack.fleet.agents.fleet_server.hosts setting supported? It
> //wasn't listed in https://github.com/elastic/observability-docs/issues/758.
> 

Author: ```suggestion
```

==== Preconfiguration settings

//QUESTION: We should add some details here to help users understand
//when they might need to modify these settings? Are these customizations that

> //most users won't need to know about?

Reviewer: these settings let users pre-define integrations and agent policies they want Fleet to load up by default, so that they do not have to manually do them in the UI. we added these to enable Cloud to configure Cloud-specific policies. IMO this is an advanced case that most users won't use

[cols="2*<a"]
|===

| `xpack.fleet.packages`

>   | List of integration packages that are installed when the {fleet} app starts

Reviewer: ```suggestion
  | List of integrations that are installed when the {fleet} app starts
```

  up for the first time. Required properties are:
  

>   `name`:: Package name.

Reviewer: ```suggestion
  `name`:: Name of the integration from the package registry
```

  `version`:: Either an exact semantic version, or the keyword `latest` to fetch

>   the latest package version.

Reviewer: ```suggestion
  the latest integration version.
```

| `xpack.fleet.agentPolicies`
  | List of agent policies that are configured when the {fleet} app starts.
Required properties are:

>   `name`:: Policy name.
>   `id`:: Unique ID for this policy. The ID may be a number or string.

Reviewer: ```suggestion
  `id`:: Unique ID for this policy. The ID may be a number or string.
  `name`:: Policy name.
```

Optional properties are:
  
  `description`:: Text description of this policy.
  `namespace`:: String identifying this policy's namespace (for advanced use cases).
  `monitoring_enabled`:: A list of either or both the keywords `logs` or `metrics`.
  `is_managed`:: If `true`, this policy is editable by the user and can only
  be changed by updating the {kib} config.
  `is_default`:: If `true`, this policy is the default agent policy.
  `is_default_fleet_server`:: If `true`, this policy is the default {fleet-server} agent policy.

>   `package_policies`:: List of policy integrations to add to this policy.

Reviewer: ```suggestion
  `package_policies`:: List of integration policies to add to this policy.
```

    `name`::: (required) Name of the integration policy.

>     `package`::: (required) Object containing

Reviewer: ```suggestion
    `package`::: (required) Integration that this policy configures
```

>       `name`:::: Name of the integration package associated with this policy.

Reviewer: ```suggestion
      `name`:::: Name of the integration associated with this policy.
```

    `description`::: Text string describing this integration policy.
    `namespace`::: String identifying this policy's namespace.
    `inputs`::: Array that overrides any default input settings for this

>     policy's package. Follows the same schema as integration inputs, with the

Reviewer: ```suggestion
    integration. Follows the same schema as integration inputs, with the
```

    exception that any object in `vars` can be passed `frozen: true` in order to
    prevent that specific `var` from being edited by the user.
|===

//QUESTION: I don't think the description of monitoring_enabled is complete.

> //Can you provide more info?

Reviewer: `monitoring_enabled` reflects this UI:
![image](https://user-images.githubusercontent.com/1965714/126406838-479c04d1-64cd-4bd9-8e0c-51e23d589908.png)

this is expected to be an array like `['logs']`, `['metrics']`, or `['logs', 'metrics']`

//QUESTION: The description of package under package_policies is incomplete
//in https://github.com/elastic/observability-docs/issues/758. Can 

> //you please fill in the details.

Reviewer: added as suggestion a few lines up

> //QUESTION: The description of package under package_policies is incomplete
> //in https://github.com/elastic/observability-docs/issues/758. Can 
> //you please fill in the details.
> 

Author: ```suggestion
```

Example configuration:

//QUESTION: Can someone in dev please confirm/test this config example. It's taken
//from https://github.com/elastic/kibana/pull/96588, and I don't know if it's

> //still valid.

Reviewer: this is still valid, maybe not the best/most realistic example, but I think it'll do for now

> //QUESTION: Can someone in dev please confirm/test this config example. It's taken
> //from https://github.com/elastic/kibana/pull/96588, and I don't know if it's
> //still valid.
> 

Author: ```suggestion
```
